presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaTable.java

```diff
@@ -0,0 +1,162 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.connector.informationschema;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.metadata.MetadataUtil.TableMetadataBuilder.tableMetadataBuilder;
+import static io.prestosql.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.util.Locale.ENGLISH;
+
+public enum InformationSchemaTable
+{
+    COLUMNS("columns"),
+    TABLES("tables"),
+    VIEWS("views"),
+    SCHEMATA("schemata"),
+    TABLE_PRIVILEGES("table_privileges"),
+    ROLES("roles"),
+    APPLICABLE_ROLES("applicable_roles"),
+    ENABLED_ROLES("enable_roles");
+
+    public static final String INFORMATION_SCHEMA = "information_schema";
+
+    private final SchemaTableName schemaTableName;
+    private final ConnectorTableMetadata tableMetadata;
+
+    InformationSchemaTable(String tableName)
+    {
+        schemaTableName = new SchemaTableName(INFORMATION_SCHEMA, tableName);
+        if (tableName.equals("columns")) {
+            tableMetadata = tableMetadataBuilder(schemaTableName)
+                    .column("table_catalog", createUnboundedVarcharType())
+                    .column("table_schema", createUnboundedVarcharType())
+                    .column("table_name", createUnboundedVarcharType())
+                    .column("column_name", createUnboundedVarcharType())
+                    .column("ordinal_position", BIGINT)
+                    .column("column_default", createUnboundedVarcharType())
+                    .column("is_nullable", createUnboundedVarcharType())
+                    .column("data_type", createUnboundedVarcharType())
+                    .column("comment", createUnboundedVarcharType())
+                    .column("extra_info", createUnboundedVarcharType())
+                    .hiddenColumn("column_comment", createUnboundedVarcharType()) // MySQL compatible
+                    .build();
+            return;
+        }
+        if (tableName.equals("tables")) {
+            tableMetadata = tableMetadataBuilder(schemaTableName)
```

Comment:
maybe tableMetadata could be a parameter to constructor?

Reply:
I thought about this but found it clumsy to do so. For example, `tableMetadataBuilder` takes the `schemaTableName` as an argument. It looks cleaner to have it initialized firstly. So should we just leave the building inside the constructor?